src/lib.rs

```diff
@@ -446,6 +446,40 @@ use {FromZeros as FromZeroes, IntoBytes as AsBytes, Ref as LayoutVerified};
 ///
 /// This derive cannot currently be applied to unsized structs without an
 /// explicit `repr` attribute.
+///
+/// Some invocations of this derive run afoul of a [known bug] in Rust's type
+/// privacy checker. For example, this code:
+///
+/// ```compile_fail,E0446
+/// use zerocopy::*;
+/// # use zerocopy_derive::*;
+///
+/// #[derive(KnownLayout)]
+/// #[repr(C)]
+/// pub struct PublicType(PrivateType);
+///
+/// #[derive(KnownLayout)]
+/// struct PrivateType;
+/// ```
+///
+/// ...results in a compilation error:
+///
+/// ```text
+/// error[E0446]: private type `PrivateType` in public interface
+///  --> examples/bug.rs:3:10
+///   |
+/// 3 | #[derive(KnownLayout)]
+///   |          ^^^^^^^^^^^ can't leak private type
+/// ...
+/// 8 | struct PrivateType;
+///   | ------------------- `PrivateType` declared as private
+///   |
+///   = note: this error originates in the derive macro `KnownLayout` (in Nightly builds, run with -Z macro-backtrace for more info)
+/// ```
+///
+/// To work around this bug, mark private field types as `pub` and annotate them with `#[doc(hidden)]`.
```

Comment:
This is only required under some circumstances. I'd double check the KnownLayout derive to get the exact list, but off the top of my head:
- It only applies to the trailing field, so I'd update the example to have two fields
- There is no separate trailing field type if it's generic (ie, if its type is `T` where `T` is a type parameter - this doesn't apply for, eg `Foo<T>`, since `Foo` could still be a private type)
- There might be reprs for which this isn't a problem? Idr.